### PR TITLE
bosh: commonly used commands

### PR DESCRIPTION
**Pull request by gkspranger**
cheat to remember commonly used bosh commands

### CODE REVIEW

File: bosh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+tags: [ bosh ]
+---
+# To list all BOSH deployments by name
```
Comment (EmilySeville7cfg) [inline bosh:4]:
```suggestion
# To list all BOSH deployments by name:
```

Optional fix: add missing `:`.

File: bosh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+bosh deps --column=name
+
+# To list a BOSH deployment's VMs, showing vital resource usage
```
Comment (EmilySeville7cfg) [inline bosh:7]:
```suggestion
# To list a BOSH deployment's VMs, showing vital resource usage
```
The same issue as above.

File: bosh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+bosh -d <deployment> vms --vitals
+
+# To SSH into a BOSH deployment VM
```
Comment (EmilySeville7cfg) [inline bosh:10]:
```suggestion
# To SSH into a BOSH deployment VM:
```

The same issue as above.

File: bosh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+bosh -d <deployment> ssh <instance-group>/(<instance-guid> | <instance-index>)
+
+# To run a shell command, via SSH, on all BOSH deployment VMs of an instance group
```
Comment (EmilySeville7cfg) [inline bosh:13]:
```suggestion
# To run a shell command, via SSH, on all BOSH deployment VMs of an instance group:
```
The same issue as above.

File: bosh [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+bosh -d <deployment> ssh <instance-group> -c '<shell-command>'
+
+# To list a BOSH deployment's errands
```
Comment (EmilySeville7cfg) [inline bosh:16]:
```suggestion
# To list a BOSH deployment's errands:
```
The same issue as above.

File: bosh [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+bosh -d <deployment> errands
+
+# To run a BOSH deployment errand
+# .. on all VMs that have colocated errand
```
Comment (EmilySeville7cfg) [inline bosh:20]:
```suggestion
# To run a BOSH deployment errand
# .. on all VMs that have colocated errand:
```
The same issue as above.

File: bosh [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+bosh -d <deployment> run-errand <errand-name>
+# .. on a specific VM
```
Comment (EmilySeville7cfg) [inline bosh:22]:
```suggestion
# .. on a specific VM:
```

The same issue as above.

File: bosh [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+bosh -d <deployment> run-errand <errand-name> --instance <instance-group>/(<instance-guid> | <instance-index>)
+
+# To list a BOSH deployment's tasks
```
Comment (EmilySeville7cfg) [inline bosh:25]:
```suggestion
# To list a BOSH deployment's tasks:
```
The same issue as above.

File: bosh [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+bosh -d <deployment> tasks
+bosh -d <deployment> tasks --recent
+
+# To show a BOSH deployment's task status and start tracking its output
```
Comment (EmilySeville7cfg) [inline bosh:29]:
```suggestion
# To show a BOSH deployment's task status and start tracking its output:
```
The same issue as above.

File: bosh [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+bosh -d <deployment> task <task-id>
+
+# To recreate a BOSH deployment's VM
+# .. safely drained and to last successful deployment state
```
Comment (EmilySeville7cfg) [inline bosh:33]:
```suggestion
# To recreate a BOSH deployment's VM
# .. safely drained and to last successful deployment state:
```
The same issue as above.

File: bosh [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+bosh -d <deployment> recreate <instance-group>/(<instance-guid> | <instance-index>)
+# .. aggressively removed, ignoring bad agents and to current deployment state
```
Comment (EmilySeville7cfg) [inline bosh:35]:
```suggestion
# .. aggressively removed, ignoring bad agents and to current deployment state:
```
The same issue as above.

File: bosh [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+tags: [ bosh ]
+---
+# To list all BOSH deployments by name
+bosh deps --column=name
+
+# To list a BOSH deployment's VMs, showing vital resource usage
+bosh -d <deployment> vms --vitals
+
+# To SSH into a BOSH deployment VM
+bosh -d <deployment> ssh <instance-group>/(<instance-guid> | <instance-index>)
```
Comment (EmilySeville7cfg) [inline bosh:11]:
```suggestion
bosh -d <deployment> ssh <instance-group>/(<instance-guid>|<instance-index>)
```
Optional fix: extra spaces can be removed.

File: bosh [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+# To list a BOSH deployment's tasks
+bosh -d <deployment> tasks
+bosh -d <deployment> tasks --recent
+
+# To show a BOSH deployment's task status and start tracking its output
+bosh -d <deployment> task <task-id>
+
+# To recreate a BOSH deployment's VM
+# .. safely drained and to last successful deployment state
+bosh -d <deployment> recreate <instance-group>/(<instance-guid> | <instance-index>)
```
Comment (EmilySeville7cfg) [inline bosh:34]:
```suggestion
bosh -d <deployment> recreate <instance-group>/(<instance-guid>|<instance-index>)
```
The same issue as above.

**Review by EmilySeville7cfg — APPROVED** [review]
No mandatory fixes found, approved.
Thanks for a contribution! ❤️

**Comment by chrisallenlane** [comment]
Thanks!

**Pull request merged**